httpClient/src/main/java/org/jfrog/artifactory/client/httpClient/http/HttpBuilderBase.java

```diff
@@ -287,8 +297,11 @@ private PoolingHttpClientConnectionManager createConnectionMgr() {
     private SSLContext buildSslContext() {
         SSLContext sslContext = null;
         try {
-            SSLContextBuilder sslBuilder = null;
+            SSLContextBuilder sslBuilder = sslContextBuilder;
             if (trustSelfSignCert) {
+                if (sslBuilder != null) {
+                    throw new IllegalStateException("Cannot configure both SSLContextBuilder and TrustSelfSignCert");
```

Comment:
What is the reason for this validation?

Reply:
Actually, it's not really needed, but the client user should be aware that using both can be deceitful since the `trustSelfSignCert` may override what is configured in the `sslBuilder`.